include/meta/Typeof.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <type_traits>
+
+namespace arch::meta::typeOf {
+/// @brief Helper struct of typeof() macro
+/// @brief (does nothing)
+struct Helper {
```

Comment:
It seems that this class more like implementation detail and those should be in `_details` namespace